wordpress-mobile/WordPress-Android
# Update finalize_release lane

This PR improves the `finalize_release` lane which now includes more steps. 
The PR depends on and should be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/10156.
The PR has a companion `release-toolkit` PR: https://github.com/wordpress-mobile/release-toolkit/pull/58

### To test:
1. Run `bundle install`.
2. Checkout a new `release/xx.xx` branch and push it with the `-u` option (the PR will commit and push on the current branch).
3. Run `bundle exec fastlane finalize_release`.
4. At the prompt, verify that the code detects the right version (12.8 right now) and confirm. 
5. Verify that the code pulls all the translations and run the release checks. 
6. Verify that the code updates `build.gradle` with the right version (12.8 right now). 
7. Verify that the metadata is downloaded in the `fastlane/metadata/android` folder and that the `release_notes.xml` file is created. 
8. Verify that the `12.8` tag is created and pushed.
9. Verify that the `12.8` milestone is closed and the code freeze tag removed. 
10. Verify that a new milestone is created with the correct `due date`.
11. Verify that the branch protection is removed on the `release/12.8` branch.

### Wrap up
1. Re-enable the branch protection on `release/12.8`.
2. Re-open the 12.8 milestone and re-add the code freeze tag.
3. Delete the `12.8` tag on your local repository and on GitHub.
4. Delete the test branch. 

P.S.: After the review, I'll release a new `release-toolkit` and update the reference in this branch before merging.

Update release notes:
